jvm/ca.ualberta.maple.swan.drivers/src/resources/models.swirl

```diff
@@ -199,19 +211,505 @@ bb0(%0 : $`*Array<τ_0_0>`, %1 : $`Array<τ_0_0>`, %2 : $`@thin Array<τ_0_0>.Ty
+bb0(%0 : $`*Array<τ_0_0>`):
+  arr = pointer_read %0, $`Array<τ_0_0>`
+  return arr
+}
+
+func [model] @`(extension in Swift):Swift.MutableCollection< where A: Swift.BidirectionalCollection>.reverse() -> ()` : $`()` {
+bb0(%0 : $`*Array<τ_0_0>`):
+  %ret = new $`()`
+  return %ret
+}
+
+// %0 -> optional String, %1 -> ptr to Array
+// Returns an array containing the non-nil results of calling
+// the given transformation with each element of this sequence.
+func [model] @`(extension in Swift):Swift.Sequence.compactMap<A>((A.Element) throws -> A1?) throws -> [A1]` : $`(@owned Array<τ_1_0>, @error Error)` {
+bb0(%0 : $`(@out Optional<String>, @error Error)`, %1 : $`*Sequence<String>`):
+  new_arr = new $`(@owned Array<τ_1_0>, @error Error)`
+  arr = pointer_read %1, $`Sequence<String>`
+  val = pointer_read arr, $`String`
+  pointer_write val to new_arr
+  return new_arr
+}
+
 // -------------------- Swift.ArraySlice ----------------------------
 
 // %0 -> value to write, %1 -> index, %2 -> array
 // Write array to pointer %0. Index not used.
 // Also likely returns the element, but the return value
 // isn't used in practice.
-func [model] @`Swift.ArraySlice.subscript.getter : (Swift.Int) -> A` : $`τ_0_0` {
+    func [model] @`Swift.ArraySlice.subscript.getter : (Swift.Int) -> A` : $`τ_0_0` {
 bb0(%0 : $`*τ_0_0`, %1 : $`Int`, %2: $`ArraySlice<τ_0_0>`):
   val = pointer_read %2, $`τ_0_0`
   pointer_write val to %0
   return val
 }
 
+// -------------------- Swift.Set ----------------------------
+
+// %0 -> array that stores the elements, %1 -> set type
+// initialising a set with elements
+func [model] @`Swift.Set.init(arrayLiteral: A...) -> Swift.Set<A>` : $`@owned Set<τ_0_0>` {
+bb0(%0 : $`Array<τ_0_0>`, %1 : $`@thin Set<τ_0_0>.Type`):
+  set = new $`@owned Set<τ_0_0>`
+  elem = pointer_read %0, $`τ_0_0`
+  pointer_write elem to set
+  return set
+}
+
+// %0 -> a type of set
+// initialising a set without elements
+func [model] @`Swift.Set.init() -> Swift.Set<A>` : $`@owned Set<τ_0_0>` {
+bb0(%0 : $`@thin Set<τ_0_0>.Type`):
+  %ret = new $`@owned Set<τ_0_0>`
+  return %ret
+}
+
+// %0 -> value to write, %1 -> the element that will be inserted , %2 -> set
+// insert element in the set
+func [model] @`Swift.Set.insert(__owned A) -> (inserted: Swift.Bool, memberAfterInsert: A)` : $`(Bool, @out τ_0_0)` {
+bb0(%0 : $`*τ_0_0`, %1 : $`*τ_0_0`, %2 : $`*Set<τ_0_0>`):
+  elem = pointer_read %1, $`τ_0_0`
+  set = pointer_read %2, $`Set<τ_0_0>`
+  pointer_write elem to %0
+  pointer_write [weak] elem to set
+  %ret = new $`(Bool, @out τ_0_0)`
+  return %ret
+}
+
+// %0 -> value to write, %1 -> the element that need to update, %2 -> set
+// update the value in the set
+func [model] @`Swift.Set.update(with: __owned A) -> A?` : $`@out Optional<τ_0_0>` {
+bb0(%0 : $`*Optional<τ_0_0>`, %1 : $`*τ_0_0`, %2 : $`*Set<τ_0_0>`):
+  val = pointer_read %0, $`Optional<τ_0_0>`
```

Comment:
Are you sure the value under `%0` has been initialized? Maybe only the pointer has been allocated. It's more likely you need to create a new `Optional`. Like you do for `Swift.Set.remove(A) -> A? : $@out Optional<τ_0_0>`.